src/Compilers/CSharp/Test/Syntax/Parsing/LambdaAttributeParsingTests.cs

```diff
@@ -3637,5 +3641,868 @@ void verify(string source, ParseOptions? parseOptions = null)
+                                                {
+                                                    N(SyntaxKind.IdentifierToken, "X");
+                                                }
+                                                M(SyntaxKind.CloseParenToken);
+                                            }
+                                        }
+                                    }
+                                }
+                                M(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.ExpressionStatement);
+                            {
+                                N(SyntaxKind.SimpleAssignmentExpression);
+                                {
+                                    N(SyntaxKind.IdentifierName);
+                                    {
+                                        N(SyntaxKind.IdentifierToken, "x");
+                                    }
+                                    N(SyntaxKind.EqualsToken);
+                                    N(SyntaxKind.CollectionExpression);
+                                    {
+                                        N(SyntaxKind.OpenBracketToken);
+                                        N(SyntaxKind.ExpressionElement);
+                                        {
+                                            N(SyntaxKind.NumericLiteralExpression);
+                                            {
+                                                N(SyntaxKind.NumericLiteralToken, "0");
+                                            }
+                                        }
+                                        N(SyntaxKind.CloseBracketToken);
+                                    }
+                                }
+                                M(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.Block);
+                            {
+                                N(SyntaxKind.OpenBraceToken);
+                                N(SyntaxKind.CloseBraceToken);
+                            }
+                            N(SyntaxKind.EmptyStatement);
+                            {
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/73789")]
+        public void ParseAttributeWithLambaPathological()
+        {
+            // Ensure we can parse this pathological lambda attribute code without getting into exponential time.
+            var tree = ParseTree(Resources.PathologicalLambdaAttributeCode, CSharpParseOptions.Default);
```

Comment:
>Resources.PathologicalLambdaAttributeCode [](http://example.com/codeflow?start=33&length=41)

This test case seems unnecessary and could be replaced by a test that directly hits the issue. Consider using the following instead:
```csharp
int n = 30;
var builder = new StringBuilder();
for (int i = 0; i < n; i++) builder.Append("[A(");
for (int i = 0; i < n; i++) builder.Append(")]");
builder.Append("class C { }");

var tree = ParseTree(builder.ToString(), CSharpParseOptions.Default);
```

Reply:
Done.